lib/src/configuration.rs

```diff
@@ -0,0 +1,59 @@
+#[derive(Debug, Clone)]
+pub struct Configuration {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub basic_auth: Option<BasicAuth>,
+    pub oauth_access_token: Option<String>,
+    pub bearer_access_token: Option<String>,
+    pub api_key: Option<ApiKey>,
+}
+
+// Check whether defining BasicAuth works like this or not, else revert to the basic definition commented out
+#[derive(Debug, Clone)]
+pub struct BasicAuth {
+    pub username: String,
+    pub password: Option<String>,
+}
+// pub type BasicAuth = (String, Option<String>);
+
+#[derive(Debug, Clone)]
+pub struct ApiKey {
+    pub prefix: Option<String>,
+    pub key: String,
+}
+
+impl Configuration {
+    pub fn new(
```

Comment:
Please add method-level doc comments to this and all other methods, with
- Description of method
- Param descriptions
- Description of return value
- Usage examples / doc (unit) tests

Reply:
Use the docstring `///` for every method/function, you can write the docstring in markdown and all codeblocks will be automatically executed when running the tests with `cargo test`. Make sure the whole block is working as standalone

The docstrings will automatically be used to generate the documentation at docs.rs

This way you don't need to write 100s of other tests, you will already have a test for each method/function, and your documentation is sure to be right and up-to-date if the tests pass (this is really useful for future sustainability)

Example docstring with test: https://github.com/vemonet/nanopub-rs/blob/main/lib/src/nanopub.rs#L113